Common/Data/Custom/Intrinio/IntrinioEconomicData.cs

```diff
@@ -0,0 +1,231 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Threading;
+using QuantConnect.Configuration;
+
+namespace QuantConnect.Data.Custom.Intrinio
+{
+    /// <summary>
+    ///     TRanformation available for the Economic data.
+    /// </summary>
+    public enum IntrinioDataTransformation
+    {
+        /// <summary>
+        ///     The rate of change
+        /// </summary>
+        Roc,
+
+        /// <summary>
+        ///     Rate of change from Year Ago
+        /// </summary>
+        AnnualyRoc,
+
+        /// <summary>
+        ///     The compounded annual rate of change
+        /// </summary>
+        CompoundedAnnualRoc,
+
+        /// <summary>
+        ///     The continuously compounded annual rate of change
+        /// </summary>
+        AnnualyCCRoc,
+
+        /// <summary>
+        ///     The continuously compounded rateof change
+        /// </summary>
+        CCRoc,
+
+        /// <summary>
+        ///     The level, no transformation.
+        /// </summary>
+        Level,
+
+        /// <summary>
+        ///     The natural log
+        /// </summary>
+        Ln,
+
+        /// <summary>
+        ///     The percent change
+        /// </summary>
+        Pc,
+
+        /// <summary>
+        ///     The percent change from year ago
+        /// </summary>
+        AnnualyPc
+    }
+
+    /// <summary>
+    ///     Access the massive repository of economic data from the Federal Reserve Economic Data system via the Intrinio API.
+    /// </summary>
+    /// <seealso cref="QuantConnect.Data.BaseData" />
+    public class IntrinioEconomicData : BaseData
+    {
+        private static DateTime LastIntrinoAPICall;
+
+        private readonly string _baseUrl = @"https://api.intrinio.com/historical_data.csv?";
+
+        private readonly IntrinioDataTransformation _dataTransformation;
+        private readonly string _password = Config.Get("intrinio-password");
+        private readonly string _user = Config.Get("intrinio-username");
```

Comment:
Do we need a static method to set these? I'm thinking about quantconnect backtesting users where we wouldn't be specifying this in the configuration files.